lips/lip-20.md

```diff
@@ -0,0 +1,426 @@
+
+## Capabilities Created at Registration
+
+Whenever a type `C` is registered as a currency, a
+[`MintCapability<C>`](https://github.com/libra/libra/blob/master/language/stdlib/modules/doc/Libra.md#resource-mintcapability)
+and
+[`BurnCapability<C>`](https://github.com/libra/libra/blob/master/language/stdlib/modules/doc/Libra.md#resource-burncapability),
+are created. If the currency is registered using the
+[`Libra::register_SCS_currency`](https://github.com/libra/libra/blob/master/language/stdlib/modules/doc/Libra.md#function-register_scs_currency)
+function, both of these capabilities are stored under the Treasury Compliance account at address
+`0xB1E55ED`. In the case that the [`Libra::register_currency`](https://github.com/libra/libra/blob/master/language/stdlib/modules/doc/Libra.md#function-register_currency)
+is used to register the currency both of these capabilities are returned to the caller, which may store them in other structures.
+The `MintCapability` and `BurnCapability` for each currency are [unique](https://github.com/libra/lip/blob/master/lips/lip-2.md#permissions); after
+registration of a currency no future mint and burn capabilities for the currency may be created.
+
+## On-chain list of registered currencies
+
+Whenever a currency is registered on-chain, the to-be-registered currencies currency code is added to the
+[`RegisteredCurrencies` on-chain config](https://github.com/libra/lip/blob/master/lips/lip-6.md#registered-currencies).
+The set of currency codes registered on-chain is always a set. Attempting to
+register a currency with an already taken currency code will fail. There are no
+restrictions on the format or length of currency codes, but there are
+[operational restrictions](#operational_things) that should be considered from the
+on-chain perspective when choosing a currency code.
+
+---
+# <a name="coin_value_representation">Representation of Value</a>
+---
+
+All assets in a currency `Currency` with a specific value `v` are represented on-chain as a
+Libra Coin value, which has the following form:
+
+```rust
+Libra<Currency> {
+    value: v,
+} : Libra<Currency>
+```
+
+Where the value `v` is defined in the "base units" for `Currency` as defined by
+the [`scaling_factor`](#scaling_factor) for `Currency`.
+
+It is important to note that in the system any value of type `Libra<Currency>` is considered a
+valid asset in the `Currency` currency with its value given by the `value` field.  Therefore the
+creation of any non-zero value of this type needs to be considered a privileged operation
+since it represents creation of value on-chain. Additionally, this is an if-and-only-if
+relation; a value represents a specific amount `v` in a given currency `Currency`, if, and only
+if the value has the type `Libra<Currency>`.
+
+---
+# <a name="minting_and_burning">Creation and Removal of Value</a>
```

Comment:
Agreed that adding code in will be helpful here. But, I think it's important to not go to code heavy, and I think walking a middle ground here would be beneficial. Particularly, in my thinking, since this is an informational LIP I don't want to describe "this is how it's done" but also to some extent "this is why it's done."